### ISSUE TITLE
💰Tracking Issue for Story Ad network integrations

### ISSUE THREAD

**Issue by calebcordry** [ampproject/amphtml#30769]
If you are an ad provider that would like to serve ads to Web Stories please reach out on this issue.

**Comment by couchcrew-thomas** [comment]
Hi @calebcordry.
I am the developer of the FeedAd amp-ad integration and would like to also integrate it inside web stories.

**Comment by calebcordry** [comment]
@couchcrew-thomas thanks for reaching out! Please let us know if you have any questions/issues. You can also open a new issue for your ad network if you want to discuss specific implementation details there. Looking forward to your implementation.

**Comment by couchcrew-thomas** [comment]
@calebcordry I am not sure where to start. Is there some kind of documentation on how to implement this? Or some pointers to what requirements need to be met for approval?

**Comment by calebcordry** [comment]
@couchcrew-thomas for sure, I should have already had something here. 

Story ads are built upon the AMPHTML Ads specification. For an ad network to serve ads to serve ads to AMP stories they must:

1. Serve valid [AMPHTML Ads](https://amp.dev/documentation/guides-and-tutorials/learn/intro-to-amphtml-ads/) with a couple extra [pieces of metadata](https://amp.dev/documentation/guides-and-tutorials/develop/story_ads_best_practices/?format=websites#meta-tag-data) that allows the amp runtime to render them. NOTE: historically AMP ads required cryptographic signing/validation but we are very near removing this requirement, so you can assume your creatives will work without signing.
2. [Create a fast fetch integration](https://amp.dev/documentation/guides-and-tutorials/contribute/adnetwork_integration/#creating-a-fast-fetch-integration) that will be checked into the AMP repo.

Hopefully those links are enough to help get you started. Also feel free to reach out on directly on slack `@caleb` if you are having issues.

**Comment by couchcrew-thomas** [comment]
@calebcordry : I see. So it is not possible to include any arbitrary amp-ad vendor inside an amp story because it is not possible to load 3rd party JS? We would have to create an extra implementation of our ad code explicitly for this?

**Comment by calebcordry** [comment]
That's right, any vendor can create a fast fetch implementation if they wish, but they must create one to serve story ads.

**Comment by sdbaron** [comment]
Hello @calebcordry!
I am developing the ** videonow ** amp-ad and want to integrate it into web stories.

**Comment by calebcordry** [comment]
@sdbaron great to hear! Please review the materials above and let me know if I can help.

**Comment by moneymoksh** [comment]
can i use other alternatives of google adsense to show ads on web stories? plz  help me out!

**Comment by montu1996** [comment]
Hello @calebcordry 

i'm also developing the *dailyhunt* amp-ad integration in web stories. i checked above link. but did not find doc regarding how to build  JS and testing.

can you pl help ?
